### DISCUSSION TITLE
v0.2.1+13

### DISCUSSION THREAD

**Discussion by hjiangsu** [thunder-app/thunder#474] [Announcements]
## Alpha Release [v0.2.1+13]

Wow is it that time again? Why yes it is, its time for another Thunder release! This release has a lot of minor tweaks and improvements to the overall Thunder experience.

A reminder for anyone who didn’t catch the previous announcement, Thunder is now officially available on Google Play! You should also be able to search up Thunder on the Play Store to find it. For more details about the launch, see this post: https://lemmy.world/post/1631968

If you are using the Google Play version or TestFlight, please note that it may take a bit of time before you receive the update. As always, **this is an alpha release**, so do expect bugs to pop up here and there. Now onto the update notes. This update brings a few major features:

- You can now select custom accent colours! This will allow you to choose a splash of colour which will be applied consistently across the whole app. Please note that Material You has to be disabled for the accent colours to be applied properly
- You can now customize the look of the comment indicators. There are different options to choose from to suit your personal preferences
- There are even more settings for you to customize. You can hide the FABs, hide NSFW posts, show community icons and authors on posts, and even more!
- Improvements to link previews. With this, you should be seeing more link previews and it should create a more consistent experience
- You can now manage blocked communities and users. Simply go to your profile, and there should be a gear icon for you to remove any blocked communities and users.
- You can now access user and community sidebars. If you visit a community, or a user, you can tap in the info icon to bring up a sidebar filled with all the information you could possible want
- Overall improvements to performance. This update should feel smoother and there should hopefully be less stuttering happening overall
- Finally, there has been some under-the-hood changes to allow for localization and translations for the future. Look forward to this in a future update!

We have a Matrix space if you would like to join in on discussions: https://matrix.to/#/#thunderapp:matrix.org

If you find any issues or would like to suggest features, please open up a new issue! As always, contributions are always welcomed here.

### Added
- Added swipe gesture to toggle read/unread status on posts - contribution from @micahmo
- Added option to enable/disable text post indicator on compact view - contribution from @micahmo
- Added improvements to link previews to be more stable, and to work more often - contribution from @micahmo
- Added instance icons in account selection - contribution from @micahmo
- Image viewer supports double-tap and slide zoom - contribution from @CTalvio
- Improvements to image viewer to be more reliable with gesture controls, and overall UI fixes - contribution from @CTalvio
- Added an option to disable FABs in feed/post page - contribution from @ajsosa
- Added customization of nested comment indicators - contribution from @micahmo
- Added ability to delete comment on long press - contribution from @vbh
- Improvements to CI/development workflow
- Added ability to view comment context when tapping on your own comment from profile - contribution from @ajsosa
- Added Matrix space to about page
- Added initial support for custom themes/accents
- Added haptic feedback when long pressing on a comment - contribution from @ajsosa
- Added width/height limit on comment images, and adjustments to comment button actions - contribution from @CTalvio
- Added sidebars to user profiles and community pages - contribution from @CTalvio
- Added blur to external link previews - contribution from @ajsosa
- Added account settings to manage blocked communities and users - contribution from @micahmo
- Added very basic initial support for localization
- Added ability to subscribe to communities without being logged in - contribution from @vbh
- Added options to show post author, and community icons within the feed - contribution from @sant0s12
- Added option to disable NSFW content - contribution from @ajsosa
- Added long-press action on image viewer to show image-only mode - contribution from @CTalvio
- Added subscription icon in post feed when you are subscribed to a given community - contribution from @micahmo 

### Changed
- Going back from a selected community in the sidebar will bring you back to the feed view - contribution from @micahmo
- Minor tweaks to toast notification when blocking communities - contribution from @micahmo
- Changed to default feed type to be "All" rather than "Local" - contribution from @micahmo
- Optimization improvements to comment cards and calculating published/edited time - contribution from @ajsosa
- Improved UI navigation experience when logging in - contribution from @micahmo
- Posts no longer have the reply swipe gesture
- Improved about page to add in-app navigation to lemmy community, and update to GitHub url - contribution from @micahmo
- Improvements to community navigation from links - contribution from @micahmo
- Updated README with Google Play Store links - contribution from @micahmo
- Increased relevance of default community search - contribution from @machinaeZER0
- Improved Gesture settings UI to be more clear - contribution from @CTalvio
- Improvements to tap + slide zoom gesture when previewing images - contribution from @CTalvio
- Downloaded images are now saved in a separate directory/album - contribution from @njshockey
- Material You theme setting is hidden on non-android devices
- Comment child count now counts total replies rather than total top level replies - contribution from @micahmo
- Desktop builds will always use external browser - contribution from @micahmo
- Adjusted way permissions are handled when saving media
- Adjusted swipe to dismiss on posts to not move divider alongside swipe gesture - contribution from @micahmo

### Fixed
- Fixed issue where comment thread would show spinning indicator even after all comments have been loaded - contribution from @ajsosa
- Fixed minor UI issue where the screen would switch from light-dark-light on app startup - contribution from @micahmo
- Fixed duplicate post regression - contribution from @ajsosa
- Fixed a couple of performance issues with constant widget rebuilding - contribution from @ajsosa
- Fixed swipe action icons not showing properly when on 2-column view - contribution from @ajsosa
- Fixed issue where interacting with saved posts from profile was throwing an error - contribution from @micahmo
- Fixed issue where markdown preview was not working when creating a post - contribution from @micahmo
- Fixed broken show link preview option in settings - contribution from @ajsosa
- Fixed issue where swiping on a comment would cause text to overflow on top of the comment indicators - contribution from @ajsosa
- Fixed issue where the app was preventing you from voting or saving multiple things within a short timeframe

---

As always, I would just like to thank everyone so far for the amazing support so far. Thunder is still a project that I can only work on the side, and it has been incredible so far. A lot of my time has been taken up to work on improvements to Thunder, and to maintain active and meaningful conversations to improve the overall experience for Thunder.

If you would like to show a token of appreciation, you can buy me a coffee! The donations will help support some of the development overhead costs (e.g., apple developer program for TestFlight builds) and other costs!

- https://ko-fi.com/hjiangsu
- https://en.liberapay.com/hjiangsu/

Thanks again for everything so far, and I hope you enjoy this update! Here’s some canned beans for you for reaching the end of this post 🥫🫘

<hr /><em>This discussion was created from the release <a href='https://github.com/thunder-app/thunder/releases/tag/v0.2.1%2B13'>v0.2.1+13</a>.</em>

**Comment by LoganAC34** [comment]
I'm having trouble installing this. Everytime I try to install, it just says "app not installed".

**Reply by CTalvio** [reply]
This can happen when there's an app conflict. Do you have a previous version installed, or the play store version installed?

**Reply by LoganAC34** [reply]
I have a previous version installed. I don't want to uninstall it though since I'm assuming I would have to re-login and that was a pain with 2FA. I didn't know there was a play store version through.

**Reply by CTalvio** [reply]
Then that is the problem.

Android will not let you install a conflicting version.

Uninstalling the version you do have installed is the only way to proceed.

**Reply by LoganAC34** [reply]
Other apps usually install over the original though. Is there any way to do the same thing?

**Reply by hjiangsu** [reply]
This depends on what version you currently have installed! There was a change in the app certificate between version v0.2.1+11 and v0.2.1+12 to prepare for the Play Store release.

If you are on v0.2.1+12, then you should be able to directly update without having to re-install the app. However, if you are on an older version (e.g., v0.2.1+11 and below), then the only way to proceed is to do a full re-install unfortunately. The times when you have to do a complete reinstall should be fairly rare, but this is one of the cases where it can't be avoided
